lib/active_support/cache/redis_store.rb

```diff
@@ -149,6 +149,44 @@ def fetch_multi(*names)
         fetched
       end
 
+      # Performs a CAS operation (like MemcachedStore) using a watch,
+      # ensuring an optimistic lock on the given name
+      def cas(name, options = nil)
+        options = merged_options(options)
+        key = normalize_key(name, options)
+        result = false
+        instrument(:cas, name, options) do
+          with do |c|
+            c.watch(key) do
+              entry = read_entry(key, options)
+              if entry
+                new_entry = yield entry.value
+                result = c.multi { write key, new_entry, options }
+              else
+                c.unwatch
+                result = false
```

Comment:
You don't need this as you've already initialised `result` to `false` on line 157? Or am I missing something?